src/engine/source/hlp2/src/alphanumeric.cpp

```diff
@@ -28,19 +28,16 @@ parsec::Parser<json::Json> getAlphanumericParser(const std::string& name, Stop,
             return res.value();
         }
 
-        auto begin {text.begin()};
-        std::advance(begin, index);
+        auto end = std::find_if(text.begin() + index, text.end(), [](char const& c) { return !std::isalnum(c); });
 
-        auto it = std::find_if(begin, text.end(), [](char const& c) { return !std::isalnum(c); });
+        auto endPos = end - text.begin();
 
-        if (it == text.end())
+        if (end == text.end())
         {
             json::Json alphaNumeric;
+            alphaNumeric.setString(std::string {text.substr(index, endPos - index)});
 
```

Comment:
Please check this decoder:

```yml
---
name: decoder/zz_test/0

metadata:
  description: Accept all

check: +ef_exists/wazuh.queue AND (+i_eq/wazuh.queue/49 OR +i_eq/wazuh.queue/50)

parse:
  logpar:
    - event.original: "<~test_tail/alphanumeric> %%%"
    - event.original: "### <~test_offset/alphanumeric>"
    - event.original: "### <~test_offset_and_tail/alphanumeric> %%%"

```
there is an issue with cases where the % % % characters are included. Specifically, the decoder is failing when it encounters the following two cases:

`### 123 %%%`
`123 %%%`

I will fix the parser to properly handle these cases. Once the parser is fixed, I will update the tests and the runtest if necessary.

To clarify the order in which I will make the fixes:

Fix the parser to handle cases with theoffset and tail.
Update the tests to ensure that all cases are handled correctly
If necessary, update the runtest to verify that the decoder works as expected in a real-world scenario